// libfort-hermes - Tracking File I/O library.
//
// Copyright (C) 2018-2023  Universit√©e de Lausanne
//
// This file is part of libfort-hermes.
//
// libfort-hermes is free software: you can redistribute it and/or modify it
// under the terms of the GNU General Public License as published by the Free
// Software Foundation, either version 3 of the License, or (at your option) any
// later version.
//
// libfort-hermes is distributed in the hope that it will be useful, but WITHOUT
// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
// FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
// details.
//
// You should have received a copy of the GNU General Public License along with
// libfort-hermes.  If not, see <http://www.gnu.org/licenses/>.

syntax = "proto3";

option optimize_for = LITE_RUNTIME;
option go_package = "github.com/formicidae-tracker/hermes";

package fort.hermes;

import "FrameReadout.proto";

message Version {
	uint32 vmajor = 1; // do not use major it conflict with some macros
	uint32 vminor = 2; // do not use minor it conflict with some macros
};

message Header {
	enum Type {
		Network = 0;
		File    = 1;
	}

	Version version  = 1;
	Type    type     = 2;
	string  previous = 3;
	int32   width    = 4;
	int32   height   = 5;

	repeated string extensionTypes = 6;
}

message Footer {
	string next = 1;
}

message FileLine {
	FrameReadout readout = 1;
	Footer       footer  = 2;
}
